### PR TITLE
Add unit tests for all existing code

### PR DESCRIPTION
**Pull request by alexaunter**

### CODE REVIEW

**Comment by alexaunter** [comment]
Add unit tests for all existing code
Add view method to get token account id
Run cargo fmt

File: tests/terminate.rs [hunk 1]
```diff
@@ -0,0 +1,842 @@
...
+
+    // TERMINATE
+    ft_storage_deposit(&users.dude, TOKEN_ID, &users.dude.account_id);
+    let res = e.terminate(&users.dude, lockup_index);
+    assert!(!res.is_ok());
+    assert!(format!("{:?}", res.status()).contains("Unauthorized"));
+}
+
+#[test]
+fn test_lockup_terminate_no_storage() {
```
Comment (frol) [inline tests/terminate.rs]: I believe this is what you meant to test here:

```suggestion
fn test_lockup_terminate_storage_cleanup() {
```

File: tests/deposit_whitelist.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                balance: amount,
+            },
+        ]),
+        claimed_balance: 0,
+        termination_config: None,
+    };
+    let balance: WrappedBalance = e.add_lockup(&e.owner, amount, &lockup).unwrap_json();
+    assert_eq!(balance.0, amount);
+    let lockups = e.get_account_lockups(&users.alice);
+    // not increased
```
Comment (frol) [inline tests/deposit_whitelist.rs]: I think this comment is not applicable here (I believe it was a copy-paste error)

```suggestion
```

File: tests/terminate.rs [hunk 3]
```diff
@@ -0,0 +1,842 @@
...
+        }),
+    };
+
+    let balance: WrappedBalance = e.add_lockup(&e.owner, amount, &lockup).unwrap_json();
+    assert_eq!(balance.0, amount);
+    let lockups = e.get_account_lockups(&users.alice);
+    assert_eq!(lockups.len(), 1);
+    let lockup_index = lockups[0].0;
+
+    // 2Y + Y * 2 / 3, 1/8 unlocked
```
Comment (frol) [inline tests/terminate.rs]:
```suggestion
    // 2Y + Y * 2 / 3, 1/4 unlocked
```

File: tests/terminate.rs [hunk 4]
```diff
@@ -0,0 +1,842 @@
...
+    assert_eq!(lockups[0].1.total_balance, amount);
+    assert_eq!(lockups[0].1.claimed_balance, 0);
+    assert_eq!(lockups[0].1.unclaimed_balance, amount / 4);
+
+    // Trying to claim
+    ft_storage_deposit(&users.alice, TOKEN_ID, &users.alice.account_id);
+    let res: WrappedBalance = e.claim(&users.alice).unwrap_json();
+    assert_eq!(res.0, amount / 4);
+
+    // TERMINATE, 3Y + Y / 3, 5/6 unlocked
```
Comment (frol) [inline tests/terminate.rs]:
```suggestion
    // TERMINATE, 3Y + Y / 3, 1/2 unlocked
```

BTW, I find it hard to follow the checkpoints when you jump from `2Y + Y * 2 / 3` to `3Y + Y / 3`. Given 2Y is a cliff, I would define it as cliff and always operate with `cliff + Y * 2 / 3` and `cliff + Y * 4 / 3`

**Review by frol — APPROVED** [review]
Overall, looks good to me

File: tests/terminate.rs (hunk 1 above)
Comment (alexaunter) [reply tests/terminate.rs]: Not exactly, this is the case when terminator is not registered in the token contract, and thus the transfer of unvested balance fails. Added more comments for readability and renamed to more verbose version.

File: tests/terminate.rs (hunk 4 above)
Comment (alexaunter) [reply tests/terminate.rs]: Fixed. To me it feels more natural to use absolute offsets and have a chart with lockup/vesting schedule at hand.

**Review by frol — APPROVED** [review]
